k4FWCore/include/k4FWCore/PodioDataSvc.h

```diff
@@ -93,7 +93,7 @@ class PodioDataSvc : public DataSvc {
     return DataSvc::registerObject("/Event", "/" + collName, wrapper);
   }
 
-  const podio::Frame& getEventFrame() const { return m_eventframe; }
+  podio::Frame& getEventFrame() { return m_eventframe; }
```

Comment:
How about having both overloads for const-correctness?
```suggestion
  podio::Frame& getEventFrame() { return m_eventframe; }
  const podio::Frame& getEventFrame() const { return m_eventframe; }
```

Reply:
Or potentially having a dedicated `getMutableEventFrame` (plus documentation that this might go away again)?

Reply:
Or make MarlinWrapper a friend to give access to protected functions.

Reply:
> Or make MarlinWrapper a friend to give access to protected functions.

This might be my favourite solution so far as it actively stops others from perpetrating the same kind of abuse

Reply:
In that case, what do you think about using passkey idiom to limit the scope of friendship?

Reply:
Given that we fully control MarlinWrapper, I would follow the KISS idiom.